lib/ConsumerImpl.h

```diff
@@ -159,7 +160,8 @@ class ConsumerImpl : public ConsumerImplBase,
     ConsumerStatsBasePtr consumerStatsBasePtr_;
 
    private:
-    bool waitingForZeroQueueSizeMessage;
+    volatile bool waitingForZeroQueueSizeMessage;
```

Comment:
Did you want to use `volatile` for thread-safety? The `volatile` keyword in C++ is much different from Java.

Reply:
Thank your reminder, Here is to ensure visibility on the multi-thread. I consulted the information and `volatile ` in C++ is not guaranteed.

I replaced it with `atomic_bool`, PTAL.